galley/pkg/server/monitoring.go

```diff
@@ -39,7 +41,15 @@ func StartSelfMonitoring(stop <-chan struct{}, port uint) {
 	}
 
 	mux := http.NewServeMux()
-	mux.Handle(metricsPath, promhttp.Handler())
+
+	registry := prometheus.DefaultRegisterer.(*prometheus.Registry)
+	exporter, err := ocprom.NewExporter(ocprom.Options{Registry: registry})
```

Comment:
Under what conditions would this fail?

Reply:
According to the current source code, nothing. Should I ignore it in the code? I figure it doesn't hurt to be defensive. Mixer does check the error as well, though they register it regardless.

Reply:
It's good to check. I was wondering whether we should return an error to the caller or not.